components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -442,10 +442,10 @@ PageContentAssociation ConversationDriver::HasPageContent() {
   return PageContentAssociation::HAS_CONTENT;
 }
 
-void ConversationDriver::DisconnectPageContents() {
+void ConversationDriver::SetShouldSendPageContents(bool should_send) {
```

Comment:
Since this function always calls `CleanUp`, which clears page contents, won't this get in to the state where toggling off and on the Page Contents Toggle will result in not having content?
I'm guessing the fix for that, if you still use this new/renamed function, would be to call `MaybeGeneratePageContents()` in this function (when param is `true`).

Reply:
won't be relevant anymore because we're deciding to cleanup article_text and friends before generating a response.